store/sqlstore/status_store.go

```diff
@@ -99,6 +102,98 @@ func (s SqlStatusStore) GetByIds(userIds []string) ([]*model.Status, error) {
 	return statuses, nil
 }
 
+// MySql doesn't has support for RETURNING clause or any euivalent so use transaction
+func (s SqlStatusStore) updateExpiredStatuses(t *gorp.Transaction) ([]string, error) {
+	currUnixTime := time.Now().UTC().Unix()
+	selectQuery, selectParams, _ := s.getQueryBuilder().
+		Select("UserId").
+		From("Status").
+		Where(
+			sq.And{
+				sq.Eq{"Status": model.STATUS_DND},
+				sq.Gt{"DNDEndTimeUnix": 0},
+				sq.LtOrEq{"DNDEndTimeUnix": currUnixTime},
+			},
+		).ToSql()
+	var selectedUsers []string
+	_, err := t.Select(&selectedUsers, selectQuery, selectParams...)
+	if err != nil {
+		return nil, errors.Wrap(err, "getExpiredDNDStatusesT: failed to get expired dnd statuses")
+	}
+	updateQuery, args, _ := s.getQueryBuilder().
+		Update("Status").
+		Where(
+			sq.And{
+				sq.Eq{"Status": model.STATUS_DND},
+				sq.Gt{"DNDEndTimeUnix": 0},
+				sq.LtOrEq{"DNDEndTimeUnix": currUnixTime},
+			},
+		).
+		Set("Status", sq.Expr("PrevStatus")).
+		Set("PrevStatus", model.STATUS_DND).
+		Set("DNDEndTimeUnix", 0).
+		Set("Manual", false).
+		ToSql()
+
+	if _, err := t.Exec(updateQuery, args...); err != nil {
+		return nil, errors.Wrapf(err, "failed to update statuses")
+	}
+	return selectedUsers, nil
+}
+
+func (s SqlStatusStore) UpdateExpiredDNDStatuses() ([]*model.Status, error) {
+	if s.DriverName() == model.DATABASE_DRIVER_MYSQL {
+		transaction, err := s.GetMaster().Begin()
+		if err != nil {
+			return nil, errors.Wrap(err, "CreateInitialSidebarCategories: begin_transaction")
+		}
+		defer finalizeTransaction(transaction)
+		userIds, err := s.updateExpiredStatuses(transaction)
+		if err != nil {
+			return nil, errors.Wrap(err, "CreateInitialSidebarCategories: createInitialSidebarCategoriesT")
+		}
+		if err := transaction.Commit(); err != nil {
+			return nil, errors.Wrap(err, "CreateInitialSidebarCategories: commit_transaction")
+		}
+		return s.GetByIds(userIds)
+	}
+
+	queryString, args, _ := s.getQueryBuilder().
+		Update("Status").
+		Where(
+			sq.And{
+				sq.Eq{"Status": model.STATUS_DND},
+				sq.Gt{"DNDEndTimeUnix": 0},
+				sq.LtOrEq{"DNDEndTimeUnix": time.Now().UTC().Unix()},
+			},
+		).
+		Set("Status", sq.Expr("PrevStatus")).
+		Set("PrevStatus", model.STATUS_DND).
+		Set("DNDEndTimeUnix", 0).
+		Set("Manual", false).
+		Suffix("RETURNING userId, Status").
```

Comment:
It does not seem correct to be returning only userID and status here. We are setting the returned status in the cache, which is then used for other purposes. We should be returning all the fields.